### PR TITLE
Navigation back button go to chapters page

### LINKED ISSUE #206: Replace back button with home button that redirects to challenge page
Related to https://github.com/saving-satoshi/saving-satoshi/issues/66#issuecomment-1448280976.

Let's repurpose the back button to a home button that leads to the chapters overview page. The user can still go backwards and forwards in the lesson content using the browser UI back/forwards, and the numerical lesson buttons in the nav bar.

@GBKS will create a mock up for us to implement. Once he does, we'll move this "To Do" for eng to pick it up.

### PR DESCRIPTION
**Pull request by gadabra**
in order to fix #206. I removed useRouter and replaced it with useContentRouter to improve content route management, and I altered the back button in the Navigation Component to link to the chapters page.

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
tACK! Confirmed that the back button goes to the chapters page. Once we merge, I can create another issue to replace the logo of the back button after we get it from design. Also will request review from kevin.

**Comment by GBKS** [comment]
This works for me, but I noticed the link it creates is not totally clean:

`https://saving-satoshi-git-fork-gadabra-navigatio-a732ca-saving-satoshi.vercel.app/en/chapters/chapter-1/en/chapters`

Looks like there might be a missing `/` to make the URL absolute.

image

File: hooks/useContentRoute.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { chapters } from 'content'
+
+const getPathData = (pathName: string) => {
+  const pathArray = pathName.split('/').filter((p) => p)
+  const isLessonRoute = pathArray.length === 4
+
+  const lessonId = isLessonRoute ? pathArray.pop() : undefined
+  const lang = pathArray.shift()
+
+  const chapterId = pathArray.pop()
```
Comment (afterburn) [inline hooks/useContentRoute.ts:11]: In the case of `/en/chapters` wouldn't `chapterId` be set to `chapters` using this approach?

**Comment by gadabra** [comment]
Thank you @GBKS. I now realize what I was missing with '/'. 
Thank you so much @afterburn, I have refactored 'getPathData'. I attempted to keep my old code, but now the problem is much easier to solve. I decide to use array destruction which looks much better than `const lessonId = isLessonRoute ? pathArray.pop() : undefined`

**Comment by GBKS** [comment]
Thanks for addressing my feedback, looks good now.

**Review by shaavan — COMMENTED** [review]
The code looks great, and the back button functions perfectly.

I suggest squashing the commits together for a clean commit history. One commit can introduce the new hook functions, and the next commit uses it in the codebase.

**Comment by gadabra** [comment]
Thank you @shaavan! I just tried squashing commits for the first time and it looks like a really cool feature of git.

**Comment by ecurrencyhodler** [comment]
@GBKS have we landed an logo for the back button in yet? Maybe we can squeeze it into this PR.

**Review by afterburn — APPROVED** [review]

**Pull request merged**